types/video.js/tslint.json

```diff
@@ -1,6 +1,7 @@
 {
     "extends": "@definitelytyped/dtslint/dt.json",
     "rules": {
+        "no-redundant-jsdoc-2": false,
```

Comment:
please remove this line and fix the error. thanks

Reply:
> please remove this line and fix the error. thanks

Thank you. Yeah, I've been trying to do so for a couple of hours. I cannot figure out why I can't get the error below to stop.

It is sort of out of scope of what my PR was going to do (tests were already failing before my changes), and I only added the offending line as a temporary measure so that the tests could pass. I've just now removed it.

Any ideas why the following snippets trigger the error below? Or is it ok to leave this PR as-is? The tests were failing prior to my changes, I've introduced no new failures here.

```
/**
         * Check if an element has a CSS class
         *
         * @param element
         *        Element to check
         *
         * @param classToCheck
         *        Class name to check for
         *
         * @return - True if the element had the class
         *         - False otherwise.
         *
         * @throws {Error}
         *         Throws an error if `classToCheck` has white space.
         */
        hasClass(element: Element, classToCheck: string): boolean;
```
and
```
    /**
     * Parent class for all advanced plugins.
     *
     * @fires   Player#beforepluginsetup
     * @fires   Player#beforepluginsetup:$name
     * @fires   Player#pluginsetup
     * @fires   Player#pluginsetup:$name
     * @listens Player#dispose
     * @throws  {Error}
     *          If attempting to instantiate the base {@link Plugin} class
     *          directly instead of via a sub-class.
     */
    interface Plugin extends EventedMixin {
```
Cause the following test failures:

```
npm run test video.js

> definitely-typed@0.0.3 test
> node --require source-map-support/register node_modules/@definitelytyped/dtslint/ types "video.js"

dtslint@0.0.143
The 'no-redundant-jsdoc-2' rule threw an error in '~/_projects/DefinitelyTyped-forked/types/video.js/index.d.ts':
Error: Unexpected tag kind: JSDocThrowsTag
    at checkTag (~/_projects/DefinitelyTyped-forked/node_modules/@definitelytyped/dtslint/src/rules/noRedundantJsdoc2Rule.ts:127:15)
    at cb (~/_projects/DefinitelyTyped-forked/node_modules/@definitelytyped/dtslint/src/rules/noRedundantJsdoc2Rule.ts:50:13)
    at visitNodes (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27171:24)
    at forEachChildInInterfaceDeclaration (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27858:203)
    at Object.forEachChild (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27256:37)
    at cb (~/_projects/DefinitelyTyped-forked/node_modules/@definitelytyped/dtslint/src/rules/noRedundantJsdoc2Rule.ts:55:15)
    at visitNodes (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27171:24)
    at forEachChildInBlock (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27216:12)
    at Object.forEachChild (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27256:37)
    at cb (~/_projects/DefinitelyTyped-forked/node_modules/@definitelytyped/dtslint/src/rules/noRedundantJsdoc2Rule.ts:55:15)
```

I also tried ignoring that rule inline, to no success.

Reply:
> > please remove this line and fix the error. thanks
> 
> Thank you. Yeah, I've been trying to do so for a couple of hours. I cannot figure out why I can't get the error below to stop.
> 
> It is sort of out of scope of what my PR was going to do (tests were already failing before my changes), and I only added the offending line as a temporary measure so that the tests could pass. I've just now removed it.
> 
> Any ideas why the following snippets trigger the error below? Or is it ok to leave this PR as-is? The tests were failing prior to my changes, I've introduced no new failures here.
> 
> ```
> /**
>          * Check if an element has a CSS class
>          *
>          * @param element
>          *        Element to check
>          *
>          * @param classToCheck
>          *        Class name to check for
>          *
>          * @return - True if the element had the class
>          *         - False otherwise.
>          *
>          * @throws {Error}
>          *         Throws an error if `classToCheck` has white space.
>          */
>         hasClass(element: Element, classToCheck: string): boolean;
> ```
> 
> and
> 
> ```
>     /**
>      * Parent class for all advanced plugins.
>      *
>      * @fires   Player#beforepluginsetup
>      * @fires   Player#beforepluginsetup:$name
>      * @fires   Player#pluginsetup
>      * @fires   Player#pluginsetup:$name
>      * @listens Player#dispose
>      * @throws  {Error}
>      *          If attempting to instantiate the base {@link Plugin} class
>      *          directly instead of via a sub-class.
>      */
>     interface Plugin extends EventedMixin {
> ```
> 
> Cause the following test failures:
> 
> ```
> npm run test video.js
> 
> > definitely-typed@0.0.3 test
> > node --require source-map-support/register node_modules/@definitelytyped/dtslint/ types "video.js"
> 
> dtslint@0.0.143
> The 'no-redundant-jsdoc-2' rule threw an error in '~/_projects/DefinitelyTyped-forked/types/video.js/index.d.ts':
> Error: Unexpected tag kind: JSDocThrowsTag
>     at checkTag (~/_projects/DefinitelyTyped-forked/node_modules/@definitelytyped/dtslint/src/rules/noRedundantJsdoc2Rule.ts:127:15)
>     at cb (~/_projects/DefinitelyTyped-forked/node_modules/@definitelytyped/dtslint/src/rules/noRedundantJsdoc2Rule.ts:50:13)
>     at visitNodes (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27171:24)
>     at forEachChildInInterfaceDeclaration (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27858:203)
>     at Object.forEachChild (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27256:37)
>     at cb (~/_projects/DefinitelyTyped-forked/node_modules/@definitelytyped/dtslint/src/rules/noRedundantJsdoc2Rule.ts:55:15)
>     at visitNodes (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27171:24)
>     at forEachChildInBlock (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27216:12)
>     at Object.forEachChild (~/_projects/DefinitelyTyped-forked/node_modules/typescript/lib/typescript.js:27256:37)
>     at cb (~/_projects/DefinitelyTyped-forked/node_modules/@definitelytyped/dtslint/src/rules/noRedundantJsdoc2Rule.ts:55:15)
> ```
> 
> I also tried ignoring that rule inline, to no success.

@peterblazejewicz  I'm not sure why the bot removed you. Do you have any pointers here?

Reply:
`@throws` tag is completely not supported by TS, that's the problem, which just surfaced:
https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html
Please amend your changes removing `@throws` blocks, those are so relevant for TS types here (one can disagree with this ofc)

Reply:
> `@throws` tag is completely not supported by TS, that's the problem, which just surfaced: https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html Please amend your changes removing `@throws` blocks, those are so relevant for TS types here (one can disagree with this ofc)

That's very helpful. Thanks so much @peterblazejewicz ! I pushed that fix after seeing the tests pass.